src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala

```diff
@@ -68,15 +78,23 @@ private[spark] class KubernetesSuite extends FunSuite with BeforeAndAfterAll wit
   }
 
   before {
+    appLocator = UUID.randomUUID().toString.replaceAll("-", "")
+    driverPodName = "spark-test-app-" + UUID.randomUUID().toString.replaceAll("-", "")
     sparkAppConf = kubernetesTestComponents.newSparkAppConf()
       .set("spark.kubernetes.container.image", image)
-      .set("spark.kubernetes.driver.label.spark-app-locator", APP_LOCATOR_LABEL)
-      .set("spark.kubernetes.executor.label.spark-app-locator", APP_LOCATOR_LABEL)
-    kubernetesTestComponents.createNamespace()
+      .set("spark.kubernetes.driver.pod.name", driverPodName)
+      .set("spark.kubernetes.driver.label.spark-app-locator", appLocator)
+      .set("spark.kubernetes.executor.label.spark-app-locator", appLocator)
+    if (!kubernetesTestComponents.hasUserSpecifiedNamespace) {
+      kubernetesTestComponents.createNamespace()
```

Comment:
Does deleting and recreating the same namespace make the rbac setting disappear?

Reply:
Actually if users pass in a namespace through `spark.kubernetes.test.namespace`, the namespace is assumed to exist and won't be deleted. Creation and deletion only applies to namespaces created automatically in the test suite.

Reply:
I just checked. It seems deleting a namespace also deletes the service account in the namespace.

Reply:
> I just checked. It seems deleting a namespace also deletes the service account in the namespace.

 That's true. Okay, I see. I was hoping we could use namespace:default for the rbac rules and the default SA gets created every time. But with making service account configurable, this makes sense.